### ISSUE TITLE
Support Datadog

### ISSUE THREAD

**Issue by julianvmodesto** [appscode/voyager#342]
Would be cool if there was support for Datadog as there is with Prometheus.

Maybe implemented w/ the Datadog agent as a side car?
https://www.datadoghq.com/blog/monitor-haproxy-with-datadog/

**Comment by tamalsaha** [comment]
Yeah. That's why we have `ingress.appscode.com/monitoring-agent` annotation. I have no experience using their service. If you want to send pr, you are most welcome!

**Comment by tamalsaha** [comment]
Here are some pointers on how to do this:

1. Create a version of [getExporterSidecar()](https://github.com/appscode/voyager/blob/master/pkg/ingress/create.go#L829) for Datadog.

2. Implement `Monitor` interface to connect to configure datadog agent . https://github.com/appscode/voyager/tree/master/pkg/monitor

**Comment by tamalsaha** [comment]
I forgot to tell one more thing, you should also add a file in the `api` package to define the annotations and parse them into a GO struct `DatadogSpec`. See the Prometheus version here:
https://github.com/appscode/voyager/blob/master/api/monitor.go
